In [16]:
words = open("words.txt").read().splitlines()

First we read the file in a list.
Each element in the list will be a line from the file.
So each element will be a single word in this case.

In [17]:
letters = {}
for word in words:
    key = word[0].lower()
    letters[key] = 1 if key not in letters else letters[key] + 1

We first create a dictionary which will have a letter as a key and the value will be
the amount of times it has been the first letter of a word.

For each word we take the first letter by doing word[0] then we check if the dictionary
contains it, if so we add 1 to the value otherwise we set the value to 1.

In [18]:
letters["a"]

30850

As you can see words.txt contains 30850 words that start with the letter a.

In [19]:
for key, value in letters.items():
    print(f"{key}: {value}")

2: 6
1: 8
&: 1
3: 5
4: 5
5: 3
6: 2
7: 2
8: 2
9: 2
a: 30850
b: 24332
c: 38933
d: 22778
': 33
e: 16815
f: 15870
g: 14466
h: 18662
i: 15220
y: 1740
-: 4
j: 4158
k: 6167
l: 14093
m: 25194
n: 16174
o: 15061
p: 40956
q: 3218
r: 21286
s: 50571
t: 25223
u: 23791
v: 6816
w: 11672
x: 609
z: 1822
